csharp/ql/src/semmle/code/cil/CallableReturns.qll

```diff
@@ -48,6 +50,6 @@ private predicate alwaysNotNullExpr(Expr expr) {
   alwaysNotNullMethod(expr.(StaticCall).getTarget())
   or
   forex(VariableUpdate vu | DefUse::variableUpdateUse(_, vu, expr) |
-    alwaysNotNullExpr(vu.getSource())
+    forex(Expr src | src = vu.getSource() | alwaysNotNullExpr(src))
```

Comment:
Great sleuthing. Do you think it makes sense to rename `getSource` to `getASource`?  The problem is that quite a few methods could be renamed to `getA...` for expressions, due to the multi-valued nature of optimized code?

Reply:
I did consider it, but I don't think it is worthwhile. Actually, it is `getOperand(int i)` that can have multiple values, so I wonder if we could end up in a situation where an instruction that pops two values could get mixed operands:
```
1. ldfld cond
2. brtrue.s 6:
3. ldstr "a"
4: ldstr "b"
5: br.s 8:
6: ldstr "c"
7: ldstr "d"
8. // instruction that pops two values
```
I.e., I wonder if we could end up with the instruction 7 having e.g. `getOperand(0) = "a" and getOperand(1) = "d"`.